parsers/jscript.c

```diff
@@ -2398,10 +2431,42 @@ static bool parsePrototype (tokenInfo *const name, tokenInfo *const token, state
 					skipArgumentList(method_body_token, false,
 									 vStringLength (signature) == 0 ? signature : NULL);
 				else
+				{
+					if ( isType (method_body_token, TOKEN_EQUAL_SIGN) &&
+						! isType (identifier_token, TOKEN_UNDEFINED))
+					{
+						if (prototype_tokens == NULL)
+							prototype_tokens = ptrArrayNew (NULL);
+
+						tokenInfo *const save_token = newToken ();
+						copyToken (save_token, identifier_token, true);
+						ptrArrayAdd (prototype_tokens, save_token);
+						identifier_token->type = TOKEN_UNDEFINED;
+					}
+					else if ( isType(method_body_token, TOKEN_IDENTIFIER))
+						copyToken (identifier_token, method_body_token, false);
+
 					readToken (method_body_token);
+				}
 			}
+			deleteToken (identifier_token);
+			accept_period_in_identifier(false);
 
 			int index = makeJsTag (token, JSTAG_METHOD, signature, NULL);
+
+			if (prototype_tokens != NULL)
+			{
+				for (int i=0; i<ptrArrayCount (prototype_tokens); i++)
+				{
+					identifier_token = ptrArrayItem (prototype_tokens, i);
```

Comment:
If you check ".prototype." in the gathering stage, you can make this making-tag stage simpler.